tiny_dnn/layers/deconvolutional_layer.h

```diff
@@ -228,8 +225,17 @@ class deconvolutional_layer : public layer {
 
   void forward_propagation(const std::vector<tensor_t *> &in_data,
                            std::vector<tensor_t *> &out_data) override {
-    // launch deconvolutional kernel
-    layer::backend_->deconv2d(in_data, out_data);
+    dws_.prev_out_ = in_data[0];
+    tensor_t &out  = *out_data[0];
+
+    fwd_ctx_.set_in_out(in_data, out_data);
+    fwd_ctx_.setParallelize(layer::parallelize());
+    fwd_ctx_.setEngine(layer::engine());
```

Comment:
BTW, keep in mind that `layer::engine()` should be replaced by `layer::backend_type()` once we get rid of the old backend structure